pgjdbc/src/test/java/org/postgresql/test/jdbc42/GetObject310Test.java

```diff
@@ -287,12 +314,12 @@ public void localTimestamps(ZoneId zoneId, String timestamp) throws SQLException
         //Also test that we get the correct values when retrieving the data as LocalDate objects
         assertEquals(localDateTime.toLocalDate(), rs.getObject("timestamp_without_time_zone_column", LocalDate.class));
         assertEquals(localDateTime.toLocalDate(), rs.getObject(1, LocalDate.class));
-      } finally {
-        rs.close();
+
+        assertDataTypeMismatch(rs, "timestamp_without_time_zone_column", OffsetTime.class);
+        // TODO: this should also not work, but that's an open discussion (see https://github.com/pgjdbc/pgjdbc/pull/2467):
+        // assertDataTypeMismatch(rs, "timestamp_without_time_zone_column", OffsetDateTime.class);
```

Comment:
This is the only conversion which was allowed by previous code. I am not sure if it really makes sense to add a sepcial case with a connectuion parameter. I would either:
- change this and disallow this type conversion (I don't expect anybody using this, if yes: they could quickly change)
- log a warning. If somebody is annoyed by the warning he/se can also change code

Reply:
@davecramer this is the test which I added in this PR mentioned in my reply: In an ideal world it should be enabled. But this would require a change in behaviour.

Reply:
Ah, ok, this is what I didn't understand. Thanks for the clarification

Reply:
The new code with OffsetTime does not accept wrong conversions (as it is completely new), the old code with OffsetDateTime still does work with timestamp (without TZ).

Thanks for review!